### PR TITLE
#409 Path 1 - Usage Unit/Quantity restricted to usage charges and renamed to Consumed Unit/Quantity

### PR DESCRIPTION
**Pull request by aqu-erp**
PR 1 of 2

Based on the conversation today, it was agreed that (non-pricing) quantities pertaining to purchase or SaaS charges need additional development. 

The changes in this PR to UsageQuantity and UsageUnit definitions intend to:

- Rename the column to Consumed Quantity and ConsumedUnit
- Narrow the application of UsageQuantity and UsageUnit to usage charges
- Retain flexibility to revisit quantities related to purchase charges
- Mitigate unclean or irrelevant values in the interim and allow future additive changes

### CODE REVIEW

**Comment by ijurica** [comment]
must be aligned with the PR #413

**Review by ijurica — APPROVED** [review]

**Comment by AWS-ZachErdman** [comment]
I prefer this path as I think UsageQuantity is a fundamental FinOps concept and should reflect the amount of a resource consumed

**Comment by udam-f2** [comment]
For the record, I prefer this option (path 1) as well. UsageQuantity describes something very clear - 'usage' and going with other more general names like 'ActualQuantity', 'MeasurementQuantity', or another one of the proposed options - to either provide distinct quantities OR to capture usage OR purchase quantities in a single column is not a net benefit as it confuses what the value means.

If the group feels strongly that the new option is better, I'm ok to go with it.

File: specification/columns/usagequantity.md [hunk 1]
```diff
@@ -1,8 +1,8 @@
 # Usage Quantity
 
-The Usage Quantity represents the volume of a given SKU associated with a [*resource*](#glossary:resource) or [*service*](#glossary:service) used or purchased, based on the [Usage Unit](#usageunit). Usage Quantity is often derived at a finer granularity or over a different time interval when compared to the [Pricing Quantity](#pricingquantity) (complementary to [Pricing Unit](#pricingunit)) and focuses on *resource* and *service* consumption, not pricing and cost.
+The Usage Quantity represents the volume of a given SKU associated with a [*resource*](#glossary:resource) or [*service*](#glossary:service) used, based on the [Usage Unit](#usageunit). Usage Quantity is often derived at a finer granularity or over a different time interval when compared to the [Pricing Quantity](#pricingquantity) (complementary to [Pricing Unit](#pricingunit)) and focuses on *resource* and *service* consumption, not pricing and cost.
```
Comment (marc-perreaut) [inline specification/columns/usagequantity.md:3]:
```suggestion
The Usage Quantity represents the volume of a given SKU associated with a [*resource*](#glossary:resource) or [*service*](#glossary:service) used, based on the [Usage Unit](#usageunit). Usage Quantity is often derived at a finer granularity or over a different time interval when compared to the [Pricing Quantity](#pricingquantity) (complementary to [Pricing Unit](#pricingunit)) and focuses on *resource* and *service* consumption, not pricing.
```
As a charge is about cost, I would just remove "and cost".
Though, that was part of the original text, so might not be removed.

**Comment by udam-f2** [comment]
> I am not fundamentally against this path, but I find it unfortunate to have these columns dedicated to ChargeCategory=Usage, while it could serve also use cases like Purchase, and so further similar columns might be introduced.
> 
> I agree the names UsageUnit / UsageQuantity can be misleading if the columns apply for other use cases than ChargeCategory=Usage, but I would prefer to go the path of finding a better name for the column.
> 
> I understand the intent of these columns is to get more standardized practitioner-friendly data than the PricingUnit / PricingQuantity, so in the end the columns work in pair (null if and only if the other is null). Maybe to make things simple and clear, the column names could be StandardizedPricingUnit / StandardizedPricingQuantity.

Hey @marc-perreaut, there was a long discussion/debate about this over many meetings. Ultimately, we seemed to have agreed on coming up with a separate column that either represents what you are charged for in distinct units OR a 'factor' column that can help remove the 'block' portion - which would apply to usage/purchase and possibly other ChargeCategory values. This will be discussed and introduced post v1.0. The decision that impacted v1.0 was to limit UsageQuantity to 'Usage' only and rename it to ConsumedQuantity. I've updated the PR to reflect the decision.

The worksheet that helped get agreement on the need to not merge both usage and purchase is here: https://docs.google.com/spreadsheets/d/1zA0brhrEntfWlzt5VNcNLBFnKPEiarajTF84o4ATeEw/edit#gid=1134244055
Rows 30/31/52/54/22 helped guide that decision - given that in the merged scenario, step pricing would not be able to show the distinct quantity you were charged for differently from what you used.

**Review by rileyjenk — APPROVED** [review]

**Comment by thecloudman** [comment]
I prefer this option. To me it leaves no doubt what Consumed Quantity/Unit are for and guides well for the usage by practitioners.

**Comment by jpradocueva** [comment]
The group approved this PR on Thursday, May 9, call.

**Review by mike-finopsorg — APPROVED** [review]

**Pull request merged**